CRM/Civicase/Hook/BuildForm/TokenTree.php

```diff
@@ -79,9 +86,38 @@ private function attachNewTokenTreeToForm(CRM_Core_Form &$form) {
       }
     }
     $this->reFormatCustomTokens($newTokenTree);
+    $newTokenTree = $this->reOrderTokens($newTokenTree);
     $this->addTokensToJs($newTokenTree);
   }
 
+  /**
+   * Reorder the tokens.
+   *
+   * @param array $newTokenTree
+   *   Restructured token tree.
+   *
+   * @return array
+   *   Reordered token tree.
+   */
+  private function reOrderTokens(array $newTokenTree) {
+    $reorderedTree = [];
+    $tokenTexts = [
+      self::CASE_TOKEN_TEXT,
+      self::CLIENT_TOKEN_TEXT,
+      self::CURRENT_USER_TOKEN_TEXT,
+    ];
+    $tokenTexts = array_merge($tokenTexts, $this->caseRolesTokenNames);
+    $tokenTexts[] = self::OTHER_TOKEN_TEXT;
+    foreach ($tokenTexts as $tokenText) {
+      if (!empty($newTokenTree[$tokenText])) {
+        $reorderedTree[$tokenText] = $newTokenTree[$tokenText];
+      }
+    }
+    unset($newTokenTree);
```

Comment:
Do we need this line? the `$newTokenTree` is not passed by reference and it's scope is within this function only

Reply:
yeah very nice catch, changed